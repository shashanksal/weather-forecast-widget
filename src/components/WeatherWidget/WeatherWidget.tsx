import React from 'react';

import { RootStateOrAny, useSelector } from 'react-redux'

//Material UI Components
import {
    Grid
} from '@material-ui/core';

//Custom components
import WeatherCard from '../WeatherCard';

const styles = {
    weatherWidgetContainer: {
        width: 800,
        height: "auto",
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
    }
}

const WeatherWidget = () => {

    const weatherData = useSelector((state: RootStateOrAny) => state?.weatherData);
    let data = typeof weatherData !== "undefined" ? weatherData[0] : []

    return (
        <div style={styles.weatherWidgetContainer}>
            <Grid container spacing={3}>
                {
                    data.length > 0 ?
                        data.slice(1, 5).map((weather: any, i: number) => {
                            return (
                                <Grid item md={3} key={i}>
                                    <WeatherCard {...weather} key={i} />
                                </Grid>
                            )
                        })
                        :
                        <div></div>
                }
            </Grid>
        </div>
    )

}

export default WeatherWidget;