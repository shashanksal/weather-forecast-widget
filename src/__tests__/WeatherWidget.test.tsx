import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import WeatherWidget from '../components/WeatherWidget';
import { Provider } from "react-redux";
import configureMockStore from 'redux-mock-store';
import thunk, { ThunkDispatch } from 'redux-thunk';

type DispatchExts = ThunkDispatch<any, void, any>;
const middlewares = [thunk];
const mockStore = configureMockStore<any, DispatchExts>(middlewares);
const mStore = mockStore();

describe('test category', () => {
	it("WeatherWidget matches snapshot" , () =>{
		const {asFragment} = render(<Provider store={mStore}><WeatherWidget /></Provider>);
		expect(asFragment()).toMatchSnapshot();
	})
})
