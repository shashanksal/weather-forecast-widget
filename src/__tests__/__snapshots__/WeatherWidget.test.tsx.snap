// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test category WeatherWidget matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 800px; height: auto; display: flex; align-items: center; justify-content: center;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
